framework/src/systems/NonlinearSystemBase.C

```diff
@@ -3456,9 +3456,12 @@ NonlinearSystemBase::computeDamping(const NumericVector<Number> & solution,
     // calling stopSolve(), it is now up to PETSc to return a
     // "diverged" reason during the next solve.
   }
-  catch (libMesh::LogicError & e)
+  catch (std::exception & e)
   {
-    // Catch the libmesh error on negative jacobian
+    // Allow the libmesh error/exception on negative jacobian
+    const std::string message = e.what();
```

Comment:
It might save a few nanoseconds to grab this by reference instead of making a copy; `mooseException()` will still make a copy later if necessary.

Reply:
done